keras/engine/training.py

```diff
@@ -1427,7 +1523,9 @@ class indices (integers) to
                               verbose=verbose, callbacks=callbacks,
                               val_f=val_f, val_ins=val_ins, shuffle=shuffle,
                               callback_metrics=callback_metrics,
-                              initial_epoch=initial_epoch)
+                              initial_epoch=initial_epoch,
+                              steps_per_epoch=steps_per_epoch,
+                              validation_steps=validation_steps)
 
     def evaluate(self, x, y, batch_size=32, verbose=1, sample_weight=None):
```

Comment:
Good point. Let's add data tensor support to `evaluate` and `predict` in this CL.